src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -516,8 +516,9 @@ int PeerReplayer::build_snap_map(const std::string &dir_root,
     uint64_t snap_id;
     if (is_remote) {
       if (!info.nr_snap_metadata) {
-        derr << ": snap_path=" << snap_path << " has invalid metadata in remote snapshot"
-             << dendl;
+        std::string failed_reason = "snapshot '" + snap  + "' has invalid metadata";
+        derr << ": " << failed_reason << dendl;
+        m_snap_sync_stats.at(dir_root).last_failed_reason = failed_reason;
```

Comment:
where/when does `last_failed_reason` get reset?

Reply:
`last_failed_reason` won't be included in the statuses by default. So it resets automatically and the `failure_reason` line clears from `peer status`, when the invalid metadata is removed from the remote snapshot.

Reply:
Probably I a misremembering - could you point me to the place where it gets reset?

Reply:
Makes sense now - will re-review :)